test/include/util/catalog_test_util.h

```diff
@@ -33,5 +34,102 @@ struct CatalogTestUtil {
     }
     return catalog::Schema(columns);
   }
+
+  // The same as RandomSchem but it won't generate Varchar Column
+  template <typename Random>
+  static catalog::Schema RandomSchemaNoVarchar(const uint16_t max_cols, Random *const generator,
+                                               storage::layout_version_t version = storage::layout_version_t(0)) {
+    TERRIER_ASSERT(max_cols > 0, "There should be at least 1 columm.");
+    catalog::col_oid_t col_oid(0);
+    const uint16_t num_attrs = std::uniform_int_distribution<uint16_t>(1, max_cols)(*generator);
+    std::vector<type::TypeId> possible_attr_types{
+        type::TypeId::BOOLEAN, type::TypeId::TINYINT, type::TypeId::SMALLINT,  type::TypeId::INTEGER,
+        type::TypeId::BIGINT,  type::TypeId::DECIMAL, type::TypeId::TIMESTAMP, type::TypeId::DATE};
+    std::vector<bool> possible_attr_nullable{true, false};
+    std::vector<catalog::Schema::Column> columns;
+    for (uint16_t i = 0; i < num_attrs; i++) {
+      type::TypeId attr_type = *RandomTestUtil::UniformRandomElement(&possible_attr_types, generator);
+      bool attr_nullable = *RandomTestUtil::UniformRandomElement(&possible_attr_nullable, generator);
+      columns.emplace_back("col_name", attr_type, attr_nullable, col_oid++);
+    }
+    return catalog::Schema(columns, version);
+  }
+
+  // Return a random ProjectedRow for the given schema. It always include all columns.
+  // The caller needs to free the ProjectedRow later by using
+  // delete[] reinterpret_cast<byte *>(pr)
+  template <typename Random>
+  static storage::ProjectedRow *RandomInsertRow(storage::SqlTable *sql_table, const catalog::Schema &schema,
+                                                storage::layout_version_t version, Random *const generator) {
+    // insertion rows need to have all columns
+    std::vector<catalog::col_oid_t> col_oids;
+
+    for (auto &col : schema.GetColumns()) {
+      col_oids.emplace_back(col.GetOid());
+    }
+    // populate the row with random bytes
+    auto pr_pair = sql_table->InitializerForProjectedRow(col_oids, version);
+    byte *buffer = common::AllocationUtil::AllocateAligned(pr_pair.first.ProjectedRowSize());
+    storage::ProjectedRow *pr = pr_pair.first.InitializeRow(buffer);
+    PopulateRandomRow(pr, schema, pr_pair.second, generator);
+
+    // if a columns is nullable, flip a coin to set it to null
+    std::vector<bool> include{true, false};
+    for (auto &col : schema.GetColumns()) {
+      if (col.GetNullable()) {
+        // flip a coin
+        if (*RandomTestUtil::UniformRandomElement(&include, generator)) {
+          pr->SetNull(pr_pair.second.at(col.GetOid()));
+        }
+      }
+    }
+    return pr;
+  }
+
+  // It populates a projected row with random bytes. The schema must not contain VarChar
```

Comment:
You'd better have some assertions to ensure the schema does not contain VarChar.